### PR TITLE
fix: Throws on `new foo?.bar!()`

### PR DESCRIPTION
**Pull request by liuxingbaoyu**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | https://github.com/babel/babel/pull/15377#discussion_r1090829448 
| Patch: Bug Fix?          | √
| Major: Breaking Change?  |
| Minor: New Feature?      |
| Tests Added + Pass?      | √
| Documentation PR Link    | 
| Any Dependency Changes?  |
| License                  | MIT

All credit to @JLHwung, thanks!

<a href="https://gitpod.io/#https://github.com/babel/babel/pull/15439"></a>

### CODE REVIEW

File: packages/babel-parser/test/fixtures/es2020/optional-chaining/new-ts-createParenthesizedExpressions-false/output.json [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+                "start":24,"end":27,"loc":{"start":{"line":2,"column":8,"index":24},"end":{"line":2,"column":11,"index":27},"identifierName":"bar"},
+                "name": "bar"
+              }
+            },
+            "arguments": []
+          },
+          "optional": true,
+          "arguments": []
+        }
+      },
```
Comment (liuxingbaoyu) [inline packages/babel-parser/test/fixtures/es2020/optional-chaining/new-ts-createParenthesizedExpressions-false/output.json:69]: There is a change here, now we keep `?.` in ast.
This aligns their behavior.
```
new foo?. bar();
new foo. bar?.();
```

**Comment by ehoogeveen-medweb** [comment]
Drive-by suggestion: Might be good to have a test for the (equally awkward) syntax `foo?.[0]` (compare `foo![0]`) as well.

File: packages/babel-parser/test/fixtures/es2020/optional-chaining/new-ts-createParenthesizedExpressions-false/input.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+new foo?.bar();
+new foo.bar?.();
+new foo?.bar!()
+
+new (foo?.bar)();
+foo?.[0]
+foo![0]
```
Comment (nicolo-ribaudo) [inline packages/babel-parser/test/fixtures/es2020/optional-chaining/new-ts-createParenthesizedExpressions-false/input.ts:7]: Shouldn't we test `new foo?.[0];` and `new foo![0];` instead?

Comment (liuxingbaoyu) [reply packages/babel-parser/test/fixtures/es2020/optional-chaining/new-ts-createParenthesizedExpressions-false/input.ts:7]: oops. :)

File: packages/babel-parser/src/parser/expression.ts [hunk 3]
```diff
@@ -1919,17 +1924,6 @@ export default abstract class ExpressionParser extends LValParser {
     node.callee = this.parseNoCallExpr();
     if (node.callee.type === "Import") {
       this.raise(Errors.ImportCallNotNewExpression, { at: node.callee });
-    } else if (
-      this.isOptionalChain(node.callee) &&
```
Comment (JLHwung) [inline packages/babel-parser/src/parser/expression.ts:1923]: Can you check if we can remove the `isOptionalChain` implementation now?

**Review by nicolo-ribaudo — APPROVED** [review]

**Pull request merged**